test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/amendment/group/AddGroupMemberContactDetailsCheckAnswersControllerSpec.scala

```diff
@@ -0,0 +1,167 @@
+
+  private def authorisedUserWithPptSubscription(): Unit =
+    authorizedUser(user =
+      newUser().copy(enrolments =
+        Enrolments(
+          Set(
+            new Enrolment(PptEnrolment.Identifier,
+                          Seq(EnrolmentIdentifier(PptEnrolment.Key, pptReference)),
+                          "activated"
+            )
+          )
+        )
+      )
+    )
+
+  private val controller = new AddGroupMemberContactDetailsCheckAnswersController(
+    authenticate = mockAuthAllowEnrolmentAction,
+    journeyAction = mockAmendmentJourneyAction,
+    mcc = mcc,
+    page = cyaPage
+  )
+
+  private val groupRegistrationInAmendment = aRegistration(
+    withGroupDetail(Some(groupDetailsWithMembers))
+  )
+
+  override protected def beforeEach(): Unit = {
+    inMemoryRegistrationAmendmentRepository.reset()
+    reset(mockSubscriptionConnector)
+    simulateGetSubscriptionSuccess(groupRegistrationInAmendment)
+    simulateUpdateSubscriptionSuccess()
+  }
+
+  "AddGroupMemberContactDetailsCheckAnswersController" when {
+
+    "displaying page" should {
+      "display successfully" when {
+        "group member present" in {
+          authorisedUserWithPptSubscription()
+          val mostRecentGroupMember = groupRegistrationInAmendment.groupDetail.get.members.last
+
+          val resp = controller.displayPage(mostRecentGroupMember.id)(getRequest())
+
+          status(resp) mustBe OK
+          contentAsString(resp) mustBe "Amend Reg - New Group Member CYA"
+        }
+      }
+      "throw IllegalStateException" when {
+        "group member is absent" in {
+          authorisedUserWithPptSubscription()
+
+          intercept[IllegalStateException] {
+            await(controller.displayPage("XXX")(getRequest()))
+          }
+        }
+      }
+    }
+
+    "details confirmed" should {
+      "update subscription at the ETMP back end" in {
+        authorisedUserWithPptSubscription()
+
+        await(controller.submit()(postRequest(JsObject.empty)))
+
+        val registrationCaptor: ArgumentCaptor[Registration] =
+          ArgumentCaptor.forClass(classOf[Registration])
+        verify(mockSubscriptionConnector).updateSubscription(any(), registrationCaptor.capture())(
+          any()
+        )
+
+        registrationCaptor.getValue mustBe groupRegistrationInAmendment
+      }
+
+      "redirect to the manage group page when update successful" in {
+        authorisedUserWithPptSubscription()
+
+        val resp = controller.submit()(postRequest(JsObject.empty))
+
+        status(resp) mustBe SEE_OTHER
+        redirectLocation(resp) mustBe Some(
+          routes.ManageGroupMembersController.displayPage().toString
+        )
+      }
+      "redirect to the post reg amend error page" when {
+        // TODO : Reinstate when we can work out why this test does not work. There's something amiss with our use of
+        //        mocks and futures!
+        "update fails due to exception being thrown" ignore {
```

Comment:
think this should be fixed in this pr

Reply:
I'll take another look. If I can't find the answer relatively quickly I think we should raise another ticket so that we can get this piece merged.

Reply:
Good call @liampan! There's actually a bug in the code under test which was causing this test to fail 🤦! Fixed.